doc/flow-definition.md

```diff
@@ -1,11 +1,308 @@
+
+### Start node
+
+Every flow has an implicit start node which represents the "landing page" i.e. the page which displays the "Start" button. There is no representation of this start node in the flow definition. Clicking the "Start" button on the "landing page" takes you to the page for the first question node (see below).
+
+Also see the documentation for [landing page templates](doc/landing-page-template.md).
+
+### Question nodes
+
+There is an implicit assumption that the question definition which appears first in the flow definition is the first question of the flow. All other "routing" between nodes is [done explicitly](doc/next-node-rules.md) in a single `next_node` block per question node.
+
+By convention, question nodes are usually listed roughly in the order that a user would visit them. Although this isn't always straightforward when there are multiple paths through the flow.
+
+Since all "routing" is done explicitly within `next_node` blocks, the order of the question definitions (other than the first one) is functionally unimportant.
+
+By convention, all question nodes are defined *before* any of the outcome nodes. However, again the order is functionally unimportant.
+
+Question nodes are defined by calls to one of the various [question-type methods](doc/question-types.md). Since the value of `self` at the "top-level" within the `#define` method is an instance of the flow, these question-type methods are defined on the `SmartAnswer::Flow` base class.
+
+For example:
+
+```ruby
+module SmartAnswer
+  class ExampleSmartAnswerFlow < Flow
+    def define
+      # metadata specified here
+
+      multiple_choice :question_key do
+        option :option_key_1
+        option :option_key_2
+
+        # optional blocks specified here
+
+        next_node do
+          # routing logic specified here
+        end
+      end
+    end
+  end
+end
+```
+
+See [In-question blocks](#in-question-blocks) below for more information on the optional blocks mentioned in a comment in the example.
+
+#### Scope
+
+The value of `self` inside the question node definition block (but outside other blocks) is the instance of the relevant question node. Thus in the example above, the calls to `#option` and `#next_node` are made on the instance of the question node.
+
+In the same way that there is a single instance of each flow class, there is only ever a single instance of each question definition in the system at any one time i.e. a flow instance has an invariant set of question node definition instances.
+
+Since the same instance of a question definition is used across multiple requests, it's important that no request-specific state is stored on them.
+
+The value of `self` inside the `next_node` blocks (or the other optional blocks) is an instance of `SmartAnswer::State` ([see below](#state)), i.e. *not* an instance of a question node.
+
```

Comment:
I mean the in-question blocks. I've changed the wording and linked it to that section in a fix-up commit: 26cc2a126cea49dc739da4cd388eb8fdc852a019.